lib/collection/src/profiling/slow_requests_collector.rs

```diff
@@ -45,6 +47,12 @@ pub struct RequestsCollector {
 const MAX_REQUESTS_LOGGED: usize = 32;
 const QUEUE_CAPACITY: usize = 64;
 
+/// Rate-limit interval for warning logs (seconds)
+const WARN_INTERVAL_SECS: u64 = 10;
+
+/// Last time a send warning was emitted (unix seconds)
+static LAST_SEND_WARN: AtomicU64 = AtomicU64::new(0);
```

Comment:
Why not `Instant::now()` and `last.elapse() > Duration::from_secs(10)`?

Just to skip a lock, as we prefer not to lock at this stage?

Reply:
`last.elapse()` would need to have Instannt type, which is not atomic